onnxscript/rewriter/ort_fusions/fused_matmul_rule_sets.py

```diff
@@ -309,14 +312,16 @@ def check(self, context, x, y, transposed: ir.Value, **_) -> orp.MatchResult:
         check_result = orp.MatchResult()
         transpose_node = _get_node(transposed, "Transpose")
         perm = _get_ints_or_default(transpose_node, "perm")
-        # If perm is not defined, the default transpose behavior is to swap
-        #    the last two dimensions, which is the correct permutation.
         if perm:
             # Check that last two dimensions are swapped
             expected_perm = list(range(len(perm)))
             expected_perm[-2], expected_perm[-1] = expected_perm[-1], expected_perm[-2]
             if perm != expected_perm:
                 return check_result.fail("Permutation values for Transpose are not correct.")
+        elif not (self._pos == 1 and _ir_utils.has_rank(x, 2)) and (self._pos == 2 and _ir_utils.has_rank(y, 2)):
+            # If perm is not defined, the default transpose behavior is to swap
+            #   all dimensions, which is correct for MatMul with rank = 2.
+            return check_result.fail("Permutation values for Transpose are not correct.")
```

Comment:
Maybe clarify what "not correct" means for users? E.g. Rank is not 2 or something like that